### PR TITLE
schedule graph sync workflow

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

Depends on: https://github.com/thoth-station/thoth-application/pull/480

Related: https://github.com/thoth-station/thoth-application/issues/398

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Function calls for new graph-sync workflow

### CODE REVIEW

File: thoth/common/workflows.py [hunk 1]
```diff
@@ -857,3 +857,28 @@ def submit_build_analysis(
         )
 
         return workflow_id
+
+    def submit_graph_sync(
+        self,
+        template_parameters: Optional[Dict[str, str]] = None,
+        workflow_parameters: Optional[Dict[str, Any]] = None,
+    ) -> Optional[str]:
+        """Submit build analysis workflow."""
```
Comment (harshad16) [inline thoth/common/workflows.py:866]: shouldn't this be ?
```suggestion
        """Submit graph sync workflow."""
```

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -1514,6 +1514,35 @@ def schedule_security_indicator(
...
         )
 
+    def schedule_graph_sync(
+        self,
+        document_id: str,
+        force_sync: bool = False,
+        *,
+        job_id: Optional[str] = None,
+    ) -> Optional[str]:
+        """Schedule a security indicator run."""
```
Comment (harshad16) [inline thoth/common/openshift.py:1524]: shouldn't this be?
```suggestion
        """Schedule a graph sync run."""
```

File: thoth/common/openshift.py [hunk 3]
```diff
@@ -1514,6 +1514,35 @@ def schedule_security_indicator(
...
+        if not self.middletier_namespace:
+            raise ConfigurationError(
+                "Unable to schedule security-indicator without middletier namespace being set"
```
Comment (harshad16) [inline thoth/common/openshift.py:1527]: Is this specific to security-indicator only ?

**Review by harshad16 — COMMENTED** [review]
just few minor questions?

**Review by harshad16 — APPROVED** [review]
lgtm :+1: 
thanks

**Review by fridex — APPROVED** [review]
/approve

thanks! :+1:

**Pull request merged**